datagrid-openshift/added/probes/probe/jdg/jolokia.py

```diff
@@ -87,6 +88,60 @@ def evaluate(self, results):
                 status.add(Status.FAILURE)
         return (min(status), messages)
 
+class CacheManagerTest(Test):
+    """
+    Checks that all defined caches are running.
+    """
+
+    def __init__(self):
+        super(CacheManagerTest, self).__init__(
+            {
+                "type": "read",
+                "attribute": [ "definedCacheCount", "createdCacheCount", "runningCacheCount" ],
+                "mbean": "jboss.infinispan:type=CacheManager,name=\"clustered\",component=CacheManager"
+            }
+        )
+
+    def evaluate(self, results):
+        """
+        Evaluates the test:
+            READY for definedCacheCount = createdCacheCount = runningCacheCount
+            NOT_READY for definedCacheCount > createdCacheCount 
+            FAILURE for definedCacheCount = createdCacheCount > runningCacheCount
+            FAILURE if the query itself failed, or all other states (STOPPING or TERMINATED)
+        """
+
+        if results["status"] != 200:
+            return (Status.FAILURE, "Jolokia query failed")
+
+	if not results["value"]:
+            return (Status.FAILURE, "No CacheManager attributes")
+
+        status = set()
+        messages = {}
+        messages["results"] = results["value"]
+        for key, value in results["value"].items():
+            if key == "createdCacheCount":
+		createdCacheCount = value
+		messages["createdCacheCount"] = createdCacheCount
+	    elif key == "definedCacheCount":
+		definedCacheCount = value
+		messages["definedCacheCount"] = definedCacheCount
+	    elif key == "runningCacheCount":
+		runningCacheCount = value
+		messages["runningCacheCount"] = runningCacheCount
+  
+        if runningCacheCount == 0:
```

Comment:
Seems like this would prevent the pod from getting bounced if none of the caches failed to start.  We'd never get to the logic that tests created.  Also, isn't this handled by default if definedCacheCount == 0?